specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,315 @@
+# Background
+Our end developers have pointed out gaps in the existing CoreWebView2.ExecuteScript method, and it is necessary
+to provide a new method to let our end developers get more information in a more convenient manner.
+The new ExecuteScriptWithResult method will provide exception information if the executed script
+failed, and provides a new method to try to get the script execution result as a string rather than as JSON
+in order to make it more convenient to interact with string results.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+We propose extending `CoreWebView2` to provide an `ExecuteScriptWithResult` 
+method. The method acts like ExecuteScript, but returns a CoreWebView2ExecuteScriptResult object that can be used to
+get the script execution result as a JSON string or as a string value if execution succeeds, and can be used to get the exception when
+execution failed.
+
+# Examples
+The following code snippets demonstrate how the ExecuteScriptWithResult can be used:
+## Win32 C++
+``` cpp
+// Tools function to generate the script code
+// Using std::wstringstream to generate script code,
+// it will generate the code like
+// 'let str_0 = "This is a string"; let n_0= str_0.replace(/a/i, "microsoft"); n_0;'
+static std::wstring GenerateScriptCode(LPCWSTR str, LPCWSTR reg, LPCWSTR item)
+{
+    if (str == nullptr || reg == nullptr || item == nullptr)
+    {
+        return L"";
+    }
+
+    // This variable is used to ensure that the
+    // variables in the script are unique.
+    static int idx = 0;
+
+    static std::wstringstream sw;
```

Comment:
Sample: this should probably not be static.

Reply:
Since the script is running in the top-level document rendered in the WebView, if the variable cannot be guaranteed to be unique, such as `let str = "..."` for each execution, it will fail from the second time due to variable redefinition, so use a static index to ensure that the variable name is unique for each execution.

And I would think it's not necessary to make the function thread-safe, so it should be no need to add thread_loacl here.

Reply:
Rather than having unique names for variables, you can run the script in an anonymous function so the scope of the variables is limited to that anonymous function and you can run it many times without worrying about unique names:

For instance, the following can be run multiple times and you don't need to worry about unique names:
```
(() => {
  let str = "...";
  let n = str.replace(..., ...);
  return n;
})();
```

Reply:
This should no longer need to be static

Reply:
This should no longer need to be static